experimental/Rings/WeierstrassPreparationTheorem.jl

```diff
@@ -0,0 +1,240 @@
+##############################################################
+#
+#Conversion from Singular (weierstr.lib)
+#
+#############################################################
+
+export weierstrass_preparation_theorem
+
+export weierstrass_division_theorem
+
+
+"""
+`order_of_generality(f::MPolyElem, n::Int)`
+
+compute for a given polynomial `f` an integer `d` such that `f` is `x_n`-general of order `d` 
```

Comment:
General rules of grammar apply for docstrings, too (so e.g. sentences are capitalized and end with a period, question mark or exclamation mark),
```suggestion
Compute for a given polynomial `f` an integer `d` such that `f` is `x_n`-general of order `d` 
```

Reply:
What does `` `x_n`-general `` mean?

Reply:
Textbook definition:
A power series f \in CC{x_1,..,x_n} is called x_n-general of order d if f(0,..,0,x_n) has a zero of order d at x_n=0.